src/main/java/org/example/custom_collections/CustomArrayList.java

```diff
@@ -0,0 +1,88 @@
+package org.example.custom_collections;
+
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+public class CustomArrayList<E> implements Iterable<E> {
+    private static final int DEFAULT_SIZE = 12;
+    private Object[] array;
+    private int size = 0;
+
+    public CustomArrayList() {
+        this.array = new Object[DEFAULT_SIZE];
+    }
+
+    public CustomArrayList(int size) {
+        if (size > 0) {
+            this.size = size;
+            this.array = new Object[size];
+        } else {
+            throw new IllegalArgumentException("Capacity must be greater than 0");
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    public E get(int index) {
+        if (index >= size || index < 0) {
```

Comment:
Why do we need size variable? What does it show?

Reply:
Size variable is needed for:
-     Managing array indices and boundaries
Variable shows how much elements inside the arrayList,